shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureWrapper.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.renderer;
+
+import android.graphics.SurfaceTexture;
+import androidx.annotation.NonNull;
+
+/**
+ * A wrapper for a SurfaceTexture that tracks whether the texture has been released.
+ *
+ * <p>The engine calls {@code SurfaceTexture.release} on the platform thread, but {@code
+ * updateTexImage} is called on the raster thread. This wrapper will prevent {@code updateTexImage}
```

Comment:
Should we add  the `@UiThread` annotation to the methods called from the platform thread? Also     `ensureRunningOnMainThread();` similar to how other methods are done in `FlutterJNI`?

Reply:
These methods do not need to be called on the platform thread.  SurfaceTextureWrapper is intended as a thread safe wrapper of SurfaceTexture.

Reply:
ah got it. Should we mention that in this comment? I'm thinking about someone who might forget about using `synchronized` or be confused about why it's needed.